src/frequenz/sdk/timeseries/_base_types.py

```diff
@@ -134,6 +134,39 @@ def map(
         )
 
 
+@runtime_checkable
+class Comparable(Protocol):
+    """
+    A protocol that requires the implementation of comparison methods.
+
+    This protocol is used to ensure that types can be compared using
+    the less than or equal to (`<=`) and greater than or equal to (`>=`)
+    operators.
+    """
+
+    def __le__(self, other: Any, /) -> bool:
+        """
+        Return True if self is less than or equal to other, otherwise False.
+
+        Args:
+            other: The value to compare with.
+
+        Returns:
+            bool: True if self is less than or equal to other, otherwise False.
+        """
```

Comment:
Suggestion: When the arguments and return can be explained in the docs summary, we don't need to be so verbose (I know a lot of code in the SDK is still very verbose, but that's because a tool we used in the past forced us to always write the whole thing). We also use "Whether ..." to document bool values in general. I recommend using this instead:
```suggestion
        """Return whether this instance is less than or equal to `other`."""
```

In case you are not aware, you can always check how the resulting docs look like by running:

```sh
pip install -e .[dev]; mkdocs serve
```

Then go to http://127.0.0.1:8000 and then you can use the search bar to look for the symbol you are interested in or navigate via the `API Reference` top menu.